src/Nethermind/Nethermind.Evm/TransactionProcessing/TransactionProcessor.cs

```diff
@@ -136,7 +136,8 @@ protected virtual TransactionResult Execute(Transaction tx, in BlockExecutionCon
             if (commit) WorldState.Commit(spec, tracer.IsTracingState ? tracer : NullTxTracer.Instance);
 
             // declare the execution witness to collect witness and also charge gas
-            IExecutionWitness executionWitness = spec.IsVerkleTreeEipEnabled ? new VerkleExecWitness(LogManager) : new NoExecWitness();
+            var verkleWorldState = WorldState as VerkleWorldState;
```

Comment:
we need some kind of error handling here or inside `VerkleExecWitness` class.
make the input nullable and then check in the constructor, if the value is null - throw an error